workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -250,11 +277,19 @@ def _get_snapshot_path(self):
     def _get_existing_plugin_names(self):
         ctx.logger.debug('Collecting existing plugins')
         existing_plugins = self._client.plugins.list(_all_tenants=True)
-        return set((p.archive_name, p['tenant_name'])
-                   for p in existing_plugins)
-
-    def _get_existing_dep_envs(self):
-        return [dep.id for dep in self._client.deployments.list()]
+        return [self._get_plugin_archive_path_id_and_tenant(p)
+                for p in existing_plugins]
+
+    def _get_plugin_archive_path_id_and_tenant(self, plugin):
+        return {
+            'path': os.path.join(
+                '/opt/manager/resources/plugins',
```

Comment:
Possibly... though I'm not convinced that it's really configurable at the moment (meanig that changing it would just cause breakages), given that things are contained within the directories that are deployed (I think) by RPM, so it's likely that we'd find unexpected assumptions based on that.

So it's doable, but not if we want to be able to merge this today, and I'm not convinced it'll bring a great deal of value without more work across the manager and probably agent repos.